### PR TITLE
XCM Fee Handling 

### PR DESCRIPTION
**Pull request by zktony**
Implemented XCM Fee Handling. Will close https://github.com/Polkadex-Substrate/thea/issues/113

### CODE REVIEW

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: xcm-simulator/src/mock_amm.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+);
+
+impl<AccountId, CurrencyId, Balance, BlockNumber> AMM<AccountId, CurrencyId, Balance, BlockNumber>
+	for MockedAMM<AccountId, CurrencyId, Balance, BlockNumber>
+{
+	fn get_amounts_out(
+		_amount_in: Balance,
+		_path: Vec<CurrencyId>,
+	) -> Result<Vec<Balance>, DispatchError> {
+		unimplemented!()
```
[the remark this answers is not in the document]
Comment (zktony) [reply xcm-simulator/src/mock_amm.rs:15]: Just to satisfy the trait.

File: xcm-simulator/src/mock_amm.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		_asset_id: CurrencyId,
+	) -> Option<(CurrencyId, CurrencyId, Pool<CurrencyId, Balance, BlockNumber>)> {
+		unimplemented!()
+	}
+
+	fn get_pool_by_asset_pair(
+		_pair: (CurrencyId, CurrencyId),
+	) -> Option<Pool<CurrencyId, Balance, BlockNumber>> {
+		unimplemented!()
+	}
```
[the remark this answers is not in the document]
Comment (zktony) [reply xcm-simulator/src/mock_amm.rs:47]: Just to satisfy the trait.

File: runtime/src/xcm_config.rs [hunk 3]
```diff
@@ -218,3 +239,139 @@ impl orml_xtokens::Config for Runtime {
...
+
+	fn buy_weight(
+		&mut self,
+		weight: u64,
+		payment: Assets,
+	) -> sp_std::result::Result<Assets, XcmError> {
+		// Calculate weight to fee
+		let fee_in_native_token =
+			T::weight_to_fee(&frame_support::weights::Weight::from_ref_time(weight));
+		// Check if
```
[the remark this answers is not in the document]
Comment (zktony) [reply runtime/src/xcm_config.rs:281]: removed

File: runtime/src/xcm_config.rs [hunk 4]
```diff
@@ -218,3 +239,139 @@ impl orml_xtokens::Config for Runtime {
 	type MaxAssetsForTransfer = MaxAssetsForTransfer;
 	type ReserveProvider = AbsoluteReserveProvider;
 }
+
+// pub struct ForeignAssetFeeHandler<T: WeightToFeeT<Balance = u128>, R: TakeRevenue, AMM: support::AMM<AccountId, u128, Balance, BlockNumber>, AC: AssetIdConverter>
+// where
```
[the remark this answers is not in the document]
Comment (zktony) [reply runtime/src/xcm_config.rs:244]: removed

**Review by Gauthamastro — APPROVED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**